sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -0,0 +1,535 @@
+     */
+    @VisibleForTesting
+    BoundedSource createSource() {
+      return new BoundedMongoDbSource(uri, database, collection, filter, numberSplit);
+    }
+
+    @Override
+    public void validate(PBegin input) {
+      Preconditions.checkNotNull(uri, "uri");
+      Preconditions.checkNotNull(database, "database");
+      Preconditions.checkNotNull(collection, "collection");
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+
+      builder.addIfNotNull(DisplayData.item("uri", uri));
+      builder.addIfNotNull(DisplayData.item("database", database));
+      builder.addIfNotNull(DisplayData.item("collection", collection));
+      builder.addIfNotNull(DisplayData.item("filter", filter));
+      builder.addIfNotNull(DisplayData.item("numberSplit", numberSplit));
+    }
+
+  }
+
+  private static class BoundedMongoDbSource extends BoundedSource {
+
+    private String uri;
+    private String database;
+    private String collection;
+    @Nullable
+    private String filter;
+    private int numberSplit;
+    private Long avgSize;
+
+    public BoundedMongoDbSource(String uri, String database, String collection, String filter,
+                                int numberSplit) {
+      this.uri = uri;
+      this.database = database;
+      this.collection = collection;
+      this.filter = filter;
+      this.numberSplit = numberSplit;
+    }
+
+    @Override
+    public Coder getDefaultOutputCoder() {
+      return SerializableCoder.of(String.class);
+    }
+
+    @Override
+    public void validate() {
+      Preconditions.checkNotNull(uri, "uri");
+      Preconditions.checkNotNull(database, "database");
+      Preconditions.checkNotNull(collection, "collection");
+    }
+
+    @Override
+    public boolean producesSortedKeys(PipelineOptions options) {
+      return false;
+    }
+
+    @Override
+    public BoundedReader createReader(PipelineOptions options) {
+      return new BoundedMongoDbReader(uri, database, collection, filter);
+    }
+
+    @Override
+    public long getEstimatedSizeBytes(PipelineOptions pipelineOptions) {
+      Long estimatedByteSize = 0L;
+      Long totalCfByteSize = 0L;
+
+      MongoClient mongoClient = new MongoClient();
+      MongoDatabase mongoDatabase = mongoClient.getDatabase(database);
+      MongoCollection mongoCollection = mongoDatabase.getCollection(collection);
+
+      // get the stats object
+      BasicDBObject stat = new BasicDBObject();
+      stat.append("collStats", collection);
+      Document stats = mongoDatabase.runCommand(stat);
+      totalCfByteSize = Long.valueOf(stats.get("size").toString());
+      avgSize = Long.valueOf(stats.get("avgObjSize").toString());
+      if (filter != null && !filter.isEmpty()) {
+        try {
+          estimatedByteSize = getFilterResultByteSize(mongoCollection, avgSize);
+        } catch (Exception e) {
+          e.printStackTrace();
```

Comment:
* Please use a logger rather than stderr.
* If we hit this branch, this function will implicitly return 0. Would be good to make this explicit (via "return 0;") if this is your intention.